sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -252,14 +243,16 @@ AZ_NODISCARD az_result az_keyvault_keys_key_get(
 
   uint8_t headers_buffer[4 * sizeof(az_pair)];
```

Comment:
This should also be an enum that the customer can change. For some services (like Storage), only the customer knows how many metadata headers they are using and so the customer must have ultimate control over this. Then we should use this enum consistently throughout the whole code base.